### ISSUE TITLE
Crash on output unplug when using the "output" feature

### ISSUE THREAD

**Issue by pedrocr** [Alexays/Waybar#410]
When I use the output feature to only display waybar on my main screen:

```
	"output": "eDP-1", // Only show on internal screen
```

I get a crash when unplugging an external HDMI screen. I do  not get this crash if Waybar is setup to display itself across all outputs. The crash is 100% reproducible and here's the backtrace on a debug build:

```
$ gdb waybar
GNU gdb (Ubuntu 8.2.91.20190405-0ubuntu3) 8.2.91.20190405-git
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from waybar...
r(gdb) run
Starting program: /usr/local/bin/waybar 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff35bb700 (LWP 26870)]
[New Thread 0x7ffff2dba700 (LWP 26874)]
[New Thread 0x7ffff258d700 (LWP 26875)]
[New Thread 0x7ffff1d8c700 (LWP 26876)]
[2019-07-13 15:20:15.867] [info] Resources files: /home/pedrocr/.config/waybar/config, /home/pedrocr/.config/waybar/style.css
[New Thread 0x7ffff13c4700 (LWP 26877)]
[New Thread 0x7ffff0bc3700 (LWP 26878)]
[New Thread 0x7fffe3fff700 (LWP 26879)]
[New Thread 0x7fffe37fe700 (LWP 26880)]
[New Thread 0x7fffe2ffd700 (LWP 26881)]
[New Thread 0x7fffe27fc700 (LWP 26882)]
[New Thread 0x7fffe1ffb700 (LWP 26883)]
[New Thread 0x7fffe17fa700 (LWP 26884)]
[New Thread 0x7fffe0ff9700 (LWP 26885)]
[New Thread 0x7fffbffff700 (LWP 26886)]
[New Thread 0x7fffbf7fe700 (LWP 26887)]
[2019-07-13 15:20:15.883] [warning] For a functionnal tray you must have libappindicator-* installed and export XDG_CURRENT_DESKTOP=Unity
[New Thread 0x7fffbeffd700 (LWP 26888)]
[2019-07-13 15:20:15.920] [info] Bar configured (width: 2560, height: 40) for output: eDP-1

Thread 1 "waybar" received signal SIGSEGV, Segmentation fault.
0x00007ffff7f9e054 in wl_proxy_marshal (proxy=0x555555b09e60, opcode=0) at ../src/wayland-client.c:780
780	../src/wayland-client.c: No such file or directory.
(gdb) bt
#0  0x00007ffff7f9e054 in wl_proxy_marshal (proxy=0x555555b09e60, opcode=0) at ../src/wayland-client.c:780
#1  0x00005555555e0527 in zxdg_output_v1_destroy(zxdg_output_v1*) (zxdg_output_v1=0x555555b09e60)
    at waybar@exe/xdg-output-unstable-v1-client-protocol.h:382
#2  0x00005555555e0bfa in waybar::Client::handleGlobalRemove(void*, wl_registry*, unsigned int)
    (data=0x55555571f010, name=40) at ../src/client.cpp:68
#3  0x00007ffff63fe81e in ffi_call_unix64 () at ../src/x86/unix64.S:76
#4  0x00007ffff63fe1ef in ffi_call (cif=cif@entry=0x7fffffffd370, fn=<optimized out>, rvalue=<optimized out>, 
    rvalue@entry=0x0, avalue=avalue@entry=0x7fffffffd440) at ../src/x86/ffi64.c:525
#5  0x00007ffff7fa112d in wl_closure_invoke
    (closure=closure@entry=0x55555580a730, flags=flags@entry=1, target=<optimized out>, 
    target@entry=0x5555558a9d40, opcode=opcode@entry=1, data=<optimized out>) at ../src/connection.c:1006
#6  0x00007ffff7f9d969 in dispatch_event (display=display@entry=0x5555557b43c0, queue=<optimized out>)
    at ../src/wayland-client.c:1427
#7  0x00007ffff7f9ee34 in dispatch_queue (queue=0x5555557b4488, display=0x5555557b43c0)
    at ../src/wayland-client.c:1573
#8  0x00007ffff7f9ee34 in wl_display_dispatch_queue_pending (display=0x5555557b43c0, queue=0x5555557b4488)
    at ../src/wayland-client.c:1815
#9  0x00007ffff7f9ee8c in wl_display_dispatch_pending (display=<optimized out>) at ../src/wayland-client.c:1878
#10 0x00007ffff6dee9a4 in _gdk_wayland_display_queue_events (display=<optimized out>)
    at ../../../../../gdk/wayland/gdkeventsource.c:201
#11 0x00007ffff6d8ead0 in gdk_display_get_event (display=display@entry=0x5555557ba0c0 [GdkWaylandDisplay])
    at ../../../../gdk/gdkdisplay.c:438
#12 0x00007ffff6dee6c2 in gdk_event_source_dispatch
    (base=<optimized out>, callback=<optimized out>, data=<optimized out>)
    at ../../../../../gdk/wayland/gdkeventsource.c:120
#13 0x00007ffff6a629ee in g_main_dispatch (context=0x5555557c7800) at ../../../glib/gmain.c:3189
#14 0x00007ffff6a629ee in g_main_context_dispatch (context=context@entry=0x5555557c7800)
    at ../../../glib/gmain.c:3854
#15 0x00007ffff6a62c88 in g_main_context_iterate
    (context=context@entry=0x5555557c7800, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>)
    at ../../../glib/gmain.c:3927
#16 0x00007ffff6a62d1c in g_main_context_iteration
    (context=context@entry=0x5555557c7800, may_block=may_block@entry=1) at ../../../glib/gmain.c:3988
#17 0x00007ffff6c64a0d in g_application_run
    (application=0x5555557ad1a0 [gtkmm__GtkApplication], argc=<optimized out>, argv=0x7fffffffe188)
    at ../../../gio/gapplication.c:2516
#18 0x00005555555e2fc5 in waybar::Client::main(int, char**) (this=0x55555571f010, argc=1, argv=0x7fffffffe188)
    at ../src/client.cpp:278
#19 0x00005555555d5770 in main(int, char**) (argc=1, argv=0x7fffffffe188) at ../src/main.cpp:22
```

**Comment by Alexays** [comment]
Is this fixed with bb99e6cf5bfaece992c82f7cd2624424e945c935?

**Comment by pedrocr** [comment]
Yep, seems to be fixed, thanks!
